zipline/src/jniMain/kotlin/app/cash/zipline/ziplineJni.kt

```diff
@@ -161,13 +165,13 @@ actual class Zipline private constructor(
 }
 
 private class RealHostPlatform(
-  val dispatcher: CoroutineDispatcher,
+  val scope: CoroutineScope,
   val jsPlatform: JsPlatform,
 ) : HostPlatform {
   private val logger = Logger.getLogger(Zipline::class.qualifiedName)
 
   override fun setTimeout(timeoutId: Int, delayMillis: Int) {
-    CoroutineScope(EmptyCoroutineContext).launch(dispatcher) {
+    scope.launch(start = UNDISPATCHED) {
```

Comment:
Without this you're subject to scheduling delay + desired delay. This should hopefully put you as close to just the desired delay as possible.